mu package does not include mu4e

It's not clear to me if this is a bug or a package request. 

> pkg install mu  

appears to not include mu4e, the popular email client for emacs users, as expected (forr mu, most distros seem to include mu4e with it), which leads to the following error within emacs when attempting to run mu4e:

> Cannot open load file: No such file or directory, mu4e. 

For Ubuntu/Debian, it *seems* mu4e may be included in the 'maildir-utils' package (https://github.com/djcb/mu). However, if a "mu" package exists in the termux pkg manager, makes me wonderr why mu4e isn't a part of it.

So I'm a bit confused. Can someone clarify what's going on here? What's the path to getting mu4e installed in termux?

I have now tried the following:

```
termux_step_pre_configure() { 
  export EMACS=$TERMUX_PREFIX/bin/emacs
}
```

However, this gives
```
/home/builder/.termux-build/mu/src/configure: line 15001: /data/data/com.termux/files/usr/bin/emacs: cannot execute binary file: Exec format error
```
which is somewhat expected, because this emacs binary is compiled for arm and cannot run on the build host. Is there a way to execute an a foreign architecture binary during the build process? :thinking: 

Alternatively, I tried `TERMUX_PKG_HOSTBUILD=true` but then I get this:
```
configure: error: Package requirements (gmime-3.0) were not met:

No package 'gmime-3.0' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables GMIME_CFLAGS
and GMIME_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.
```

Also after looking at the build scripts for other packages using gmime it was not clear to me how to fix this :thinking:

--- a/packages/mu/build.sh
+++ b/packages/mu/build.sh
@@ -3,7 +3,23 @@ TERMUX_PKG_DESCRIPTION="Maildir indexer/searcher and Emacs client (mu4e)"
 TERMUX_PKG_LICENSE="GPL-3.0"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION=1.6.9
+TERMUX_PKG_REVISION=2
 TERMUX_PKG_SRCURL=https://github.com/djcb/mu/releases/download/$TERMUX_PKG_VERSION-signed/mu-$TERMUX_PKG_VERSION.tar.xz
 TERMUX_PKG_SHA256=8d99430a4f6942eb3122008bb897c4480cda64b30ff4575c32ded86d7edbaf58
 TERMUX_PKG_AUTO_UPDATE=true
-TERMUX_PKG_DEPENDS="glib, libc++, libxapian, libgmime"
+TERMUX_PKG_DEPENDS="emacs, glib, libc++, libxapian, libgmime"
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="--disable-gtk --disable-webkit"
+
+termux_step_create_debscripts() {
+	cat <<- EOF > ./postinst
+	#!$TERMUX_PREFIX/bin/sh
+	echo "(setq byte-compile-warnings nil)" > $TERMUX_PREFIX/share/emacs/site-lisp/mu4e/nowarnings.el
+	LC_ALL=C $TERMUX_PREFIX/bin/emacs -no-site-file -q -batch -l $TERMUX_PREFIX/share/emacs/site-lisp/mu4e/nowarnings.el -f batch-byte-compile $TERMUX_PREFIX/share/emacs/site-lisp/mu4e/*.el
+	rm -f $TERMUX_PREFIX/share/emacs/site-lisp/mu4e/nowarnings.elc
+	chmod 644 $TERMUX_PREFIX/share/emacs/site-lisp/mu4e/*.elc
+	EOF
+
+	cat <<- EOF > ./prerm
+	rm -f $TERMUX_PREFIX/share/emacs/site-lisp/mu4e/*.elc
+	EOF
+}

--- a/packages/mu/mu4e-dont-bytecompile.patch
+++ b/packages/mu/mu4e-dont-bytecompile.patch
@@ -1,0 +1,44 @@
+@@ from https://sources.debian.org/patches/maildir-utils/1.6.9-1/mu4e-dont-bytecompile/
+---
+ configure.ac     |   18 ++++--------------
+ mu4e/Makefile.am |    2 +-
+ 2 files changed, 5 insertions(+), 15 deletions(-)
+
+--- a/mu4e/Makefile.am
++++ b/mu4e/Makefile.am
+@@ -21,7 +21,7 @@ SUBDIRS=
+ info_TEXINFOS=mu4e.texi
+ mu4e_TEXINFOS=fdl.texi
+ 
+-dist_lisp_LISP=			\
++dist_lisp_DATA=			\
+ 	mu4e-actions.el		\
+ 	mu4e-compose.el		\
+ 	mu4e-context.el		\
+--- a/configure.ac
++++ b/configure.ac
+@@ -85,20 +85,10 @@ AM_COND_IF(HAVE_MAKEINFO,[],[
+   AC_SUBST(MAKEINFO,[true])
+ ])
+ 
+-# we need emacs for byte-compiling mu4e
+-build_mu4e=no
+-AC_ARG_ENABLE([mu4e],
+-   AS_HELP_STRING([--disable-mu4e],[Disable building mu4e]))
+-AS_IF([test "x$enable_mu4e" != "xno"], [
+-  AM_PATH_LISPDIR
+-  AS_IF([test "x$lispdir" != "xno"], [
+-    emacs_version="$($EMACS --version | head -1)"
+-    lispdir="${lispdir}/mu4e/"
+-  ])
+-  AS_CASE([$emacs_version],
+-    [*25.3*],[build_mu4e=yes],
+-    [*26*|*27*|*28*|*29*],[build_mu4e=yes],
+-    [AC_WARN([emacs is too old to build mu4e (need emacs >= 25.3)])])
++build_mu4e=yes
++AM_PATH_LISPDIR
++AS_IF([test "x$lispdir" != "xno"], [
++  lispdir="${lispdir}/mu4e/"
+ ])
+ AM_CONDITIONAL(BUILD_MU4E, test "x$build_mu4e" = "xyes")
+ 

